File: scratch/generate_data.dart
// ignore_for_file: avoid_print
import 'dart:convert';
import 'dart:io';
import 'dart:math';

void main() {
  final rand = Random();
  final daysAr = ["السبت", "الأحد", "الإثنين", "الثلاثاء", "الأربعاء", "الخميس", "الجمعة"];
  final subjects = ["Gem", "المعاصر", "أخرى"];
  final types = ["سنتر", "أونلاين", "أخرى"];
  
  final firstNames = ["أحمد", "محمد", "محمود", "عمر", "علي", "يوسف", "حسن", "حسين", "عبد الله", "مصطفى", "إبراهيم", "فاطمة", "مريم", "سارة", "نور", "ملك", "حبيبة", "شهد", "ندى", "آية", "رؤى", "زياد", "كريم", "ياسين", "سيف", "خالد", "مازن", "سالم"];
  final lastNames = ["السباعي", "النجار", "الحداد", "المصري", "المهندس", "السيد", "محمود", "علي", "إبراهيم", "حسن", "حسين", "كمال", "سعيد", "صالح", "علام", "منصور", "زكي", "رضوان"];

  String uuid() {
    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replaceAllMapped(RegExp(r'[xy]'), (match) {
      final r = rand.nextInt(16);
      final v = match.group(0) == 'x' ? r : (r & 0x3 | 0x8);
      return v.toRadixString(16);
    });
  }

  final groups = [];
  final students = [];
  final now = DateTime.now().toIso8601String();

  for (int i = 1; i <= 15; i++) {
    final gId = uuid();
    final d1 = daysAr[rand.nextInt(daysAr.length)];
    var d2 = daysAr[rand.nextInt(daysAr.length)];
    while (d1 == d2) {
      d2 = daysAr[rand.nextInt(daysAr.length)];
    }

    final daysJson = jsonEncode([
      {"day": d1, "time": "${(rand.nextInt(8) + 12).toString().padLeft(2, '0')}:00"},
      {"day": d2, "time": "${(rand.nextInt(8) + 12).toString().padLeft(2, '0')}:00"}
    ]);

    final grades = ['الأول', 'الثاني', 'الثالث'];
    groups.add({
      "id": gId,
      "name": "المجموعة رقم $i - الصف ${grades[rand.nextInt(grades.length)]}",
      "type": types[rand.nextInt(types.length)],
      "subject": subjects[rand.nextInt(subjects.length)],
      "days": daysJson,
      "monthly_price": [150, 200, 250, 300][rand.nextInt(4)].toDouble(),
      "status": "نشطة",
      "created_at": now
    });

    final numStudents = rand.nextInt(16) + 15; // 15 to 30
    for (int j = 0; j < numStudents; j++) {
      final sId = uuid();
      final name = "${firstNames[rand.nextInt(firstNames.length)]} ${lastNames[rand.nextInt(lastNames.length)]}";
      final phone = "010${(rand.nextInt(90000000) + 10000000)}";
      final parent = "011${(rand.nextInt(90000000) + 10000000)}";

      students.add({
        "id": sId,
        "name": name,
        "phone": phone,
        "parent_phone": parent,
        "level": rand.nextInt(10) + 1,
        "group_id": gId,
        "notes": "",
        "points": rand.nextInt(50),
        "status": "نشط",
        "created_at": now
      });
    }
  }

  final data = {
    "groups": groups,
    "students": students,
    "attendance": [],
    "exams": [],
    "exam_results": [],
    "payments": []
  };

  final file = File(r"C:\Users\AM\Desktop\backup_test.json");
  file.writeAsStringSync(jsonEncode(data));
  print("Done!");
}

File: scratch/generate_data.py
import json
import random
import uuid
import datetime

def generate_data():
    days_ar = ["السبت", "الأحد", "الإثنين", "الثلاثاء", "الأربعاء", "الخميس", "الجمعة"]
    subjects = ["Gem", "المعاصر", "أخرى"]
    types = ["سنتر", "أونلاين", "أخرى"]
    
    first_names = ["أحمد", "محمد", "محمود", "عمر", "علي", "يوسف", "حسن", "حسين", "عبد الله", "مصطفى", "إبراهيم", "فاطمة", "مريم", "سارة", "نور", "ملك", "حبيبة", "شهد", "ندى", "آية", "رؤى", "زياد", "كريم", "ياسين", "سيف", "خالد", "مازن", "سالم"]
    last_names = ["السباعي", "النجار", "الحداد", "المصري", "المهندس", "السيد", "محمود", "علي", "إبراهيم", "حسن", "حسين", "كمال", "سعيد", "صالح", "علام", "منصور", "زكي", "رضوان"]

    groups = []
    students = []
    
    now = datetime.datetime.now().isoformat()
    
    for i in range(1, 16):
        g_id = str(uuid.uuid4())
        d1 = random.choice(days_ar)
        d2 = random.choice([d for d in days_ar if d != d1])
        
        days_json = json.dumps([
            {"day": d1, "time": f"{random.randint(12,20):02d}:00"},
            {"day": d2, "time": f"{random.randint(12,20):02d}:00"}
        ])
        
        groups.append({
            "id": g_id,
            "name": f"المجموعة رقم {i} - الصف {random.choice(['الأول', 'الثاني', 'الثالث'])}",
            "type": random.choice(types),
            "subject": random.choice(subjects),
            "days": days_json,
            "monthly_price": float(random.choice([150, 200, 250, 300])),
            "status": "نشطة",
            "created_at": now
        })
        
        num_students = random.randint(15, 30)
        for _ in range(num_students):
            s_id = str(uuid.uuid4())
            name = f"{random.choice(first_names)} {random.choice(last_names)}"
            phone = f"010{random.randint(10000000, 99999999)}"
            parent = f"011{random.randint(10000000, 99999999)}"
            
            students.append({
                "id": s_id,
                "name": name,
                "phone": phone,
                "parent_phone": parent,
                "level": random.randint(1, 10),
                "group_id": g_id,
                "notes": "",
                "points": random.randint(0, 50),
                "status": "نشط",
                "created_at": now
            })
            
    data = {
        "groups": groups,
        "students": students,
        "attendance": [],
        "exams": [],
        "exam_results": [],
        "payments": []
    }
    
    with open(r"C:\Users\AM\Desktop\backup_test.json", "w", encoding="utf-8") as f:
        json.dump(data, f, ensure_ascii=False, indent=2)

if __name__ == "__main__":
    generate_data()
